tests/mocks.py

```diff
@@ -96,6 +97,51 @@ def validate_change(change):
     return True
 
 
+def get_stack(_phid, phabdouble):
+    phids = set()
+    new_phids = {_phid}
+    edges = []
+
+    # Repeatedly request all related edges, adding connected revisions
+    # each time until no new revisions are found.
+    # NOTE: this was adapted from previous implementation of build_stack_graph.
+    while new_phids:
+        phids.update(new_phids)
+        edges = [
+            e
+            for e in phabdouble._edges
+            if e["sourcePHID"] in phids
+            and e["edgeType"] in ("revision.parent", "revision.child")
+        ]
+        new_phids = set()
+        for edge in edges:
+            new_phids.add(edge["sourcePHID"])
+            new_phids.add(edge["destinationPHID"])
+
+        new_phids = new_phids - phids
+
+    # Treat the stack like a commit DAG, we only care about edges going
+    # from child to parent. This is enough to represent the graph.
+    edges = {
+        (edge["sourcePHID"], edge["destinationPHID"])
+        for edge in edges
+        if edge["edgeType"] == "revision.parent"
+    }
+
+    stack_graph = defaultdict(list)
+    sources = [edge[0] for edge in edges]
```

Comment:
```suggestion
    sources = [source for source, dest in edges]
```

Reply:
FWIW, this is the same source/dest concept as in the [other comment](https://github.com/mozilla-conduit/lando-api/pull/279/files#r1132474294).